x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleActionsRegistry.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.common.util.set.Sets;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Defines a record of the actions that are allowed for the timeseries lifecycle.
+ * The order of actions is versioned, newer versions must contain all previous defined actions and are
+ * allowed to make additive changes (we do not support removing actions, but actions to be removed
+ * must be converted to no-ops)
+ */
+public final class TimeseriesLifecycleActionsRegistry {
+
+    public static final int VERSION_ONE = 1;
+    // moves the downsample action after the migrate action in the warm and cold phases
+    public static final int VERSION_TWO = 2;
+    public static final int CURRENT_VERSION = VERSION_TWO;
+
+    public static final String HOT_PHASE = "hot";
+    public static final String WARM_PHASE = "warm";
+    public static final String COLD_PHASE = "cold";
+    public static final String FROZEN_PHASE = "frozen";
+    public static final String DELETE_PHASE = "delete";
+
+    public static final Map<Integer, List<String>> ORDERED_VALID_HOT_ACTIONS = Map.of(
+        VERSION_ONE,
+        Stream.of(
+            SetPriorityAction.NAME,
+            UnfollowAction.NAME,
+            RolloverAction.NAME,
+            ReadOnlyAction.NAME,
+            DownsampleAction.NAME,
+            ShrinkAction.NAME,
+            ForceMergeAction.NAME,
+            SearchableSnapshotAction.NAME
+        ).filter(Objects::nonNull).toList(),
+        VERSION_TWO,
+        Stream.of(
+            SetPriorityAction.NAME,
+            UnfollowAction.NAME,
+            RolloverAction.NAME,
+            ReadOnlyAction.NAME,
+            DownsampleAction.NAME,
+            ShrinkAction.NAME,
+            ForceMergeAction.NAME,
+            SearchableSnapshotAction.NAME
+        ).filter(Objects::nonNull).toList()
```

Comment:
Since these are the same, can we put them in a variable so it's clear here that the order isn't changing between versions?